plugins/amazonq/chat/jetbrains-community/tst/software/aws/toolkits/jetbrains/services/amazonqFeatureDev/controller/FeatureDevControllerTest.kt

```diff
@@ -46,43 +74,72 @@ class FeatureDevControllerTest : FeatureDevTestBase() {
     private lateinit var spySession: Session
     private lateinit var featureDevClient: FeatureDevClient
 
-    private val tabId = "tabId"
+    private val newFileContents = listOf(
+        NewFileZipInfo("test.ts", "This is a comment", false),
+        NewFileZipInfo("test2.ts", "This is a rejected file", true)
+    )
+    private val deletedFiles = listOf(
+        DeletedFileInfo("delete.ts", false),
+        DeletedFileInfo("delete2.ts", true)
+    )
+    private val references = listOf(CodeReference(information = "test"))
 
     @Before
     override fun setup() {
         super.setup()
         featureDevClient = mock()
         messenger = mock()
         chatSessionStorage = mock()
+        projectRule.project.replaceService(FeatureDevClient::class.java, featureDevClient, disposableRule.disposable)
         appContext = mock<AmazonQAppInitContext> {
-            on { project }.thenReturn(project)
+            on { project }.thenReturn(projectRule.project)
             on { messagesFromAppToUi }.thenReturn(messenger)
         }
-        authController = spy(AuthController())
+        authController = mock()
```

Comment:
We should not mock and spy on the same object; You can either use `authController = mock()` or `authController = spy(AuthController())` and override the response accordingly.

Reply:
Also fixing this